packages/solana/test/models_test.dart

```diff
@@ -0,0 +1,165 @@
+// ignore_for_file: prefer-first
+
+import 'package:solana/dto.dart';
+import 'package:solana/solana.dart';
+import 'package:test/test.dart';
+
+import 'config.dart';
+
+void main() {
+  late final SolanaClient solanaClient;
+  late final Wallet source;
+  late final Wallet destination;
+  late Mint token;
+
+  setUpAll(() async {
+    source = await Ed25519HDKeyPair.random();
+    solanaClient = createTestSolanaClient();
+    destination = await Ed25519HDKeyPair.random();
+    // Add tokens to the sender
+    await solanaClient.requestAirdrop(
+      lamports: 100 * lamportsPerSol,
+      address: source.publicKey,
+      commitment: Commitment.confirmed,
+    );
+    token = await solanaClient.initializeMint(
+      mintAuthority: source,
+      decimals: 2,
+      commitment: Commitment.confirmed,
+    );
+    final associatedAccount = await solanaClient.createAssociatedTokenAccount(
+      mint: token.address,
+      funder: source,
+      commitment: Commitment.confirmed,
+    );
+
+    await solanaClient.mintTo(
+      destination: Ed25519HDPublicKey.fromBase58(associatedAccount.pubkey),
+      amount: _tokenMintAmount,
+      mint: token.address,
+      authority: source,
+      commitment: Commitment.confirmed,
+    );
+  });
+
+  group('SplTokenTransferCheckedInfo test', () {
+    test('authority is not available', () {
+      // TODO
```

Comment:
Not sure how to make `multisig` transaction. Because then `authority` will be `null` and `multisig` with `signatures` - won't. Would love to have helping hand here.